azext_iot/iothub/providers/certificate.py

```diff
@@ -0,0 +1,102 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from typing import Dict, Optional
+from azext_iot.common.embedded_cli import EmbeddedCLI
+from azext_iot.digitaltwins.common import ABORT_MSG
+from azext_iot.iothub.common import (
+    CA_REVERT_WARNING,
+    CA_TRANSITION_API_VERSION,
+    CA_TRANSITION_WARNING,
+    CONT_INPUT_MSG,
+    DEFAULT_ROOT_AUTHORITY,
+    HUB_PROVIDER,
+    NO_CHANGE_MSG,
+    CertificateAuthorityVersions
+)
+from azext_iot.iothub.providers.base import IoTHubProvider
+from azure.cli.core.azclierror import ManualInterrupt
+from knack.log import get_logger
+from knack.prompting import prompt_y_n
+
+
+logger = get_logger(__name__)
+
+
+class CertificateProvider(IoTHubProvider):
+    def __init__(
+        self,
+        cmd,
+        hub_name: Optional[str] = None,
+        rg: Optional[str] = None,
+        login: Optional[str] = None,
```

Comment:
Why support login or auth_type_dataplane on init here? I was under the impression this was control plane only functionality.

Reply:
tbh this was more for the discovery of the hub rather than actually calling the command (and me not knowing better back then). I will remove cause it is more or less pointless